job.py

```diff
@@ -55,31 +55,26 @@ class Job:
     # These are dynamic, cached, and need to be udpated periodically
     phase = (None, None)   # Phase/subphase
 
-    def get_running_jobs(logroot):
-        '''Return the list of running plot jobs, reading the process table from scratch.'''
+    def get_running_jobs(logroot, cached_jobs=None):
+        '''Return a list of running plot jobs.  If a cache of preexisting jobs is provided,
+           reuse those previous jobs without updating their information.  Always look for
+           new jobs not already in the cache.'''
         jobs = []
-        for proc in psutil.process_iter(['pid', 'name']):
-            if os.path.basename(proc.name()) == 'chia':
-                args = proc.cmdline()
-                # n.b.: args[0]=python, args[1]=chia
-                if len(args) >= 4 and args[2] == 'plots' and args[3] == 'create':
-                    jobs.append(Job(proc, logroot))
-        return jobs
-
-    def get_running_jobs_w_cache(logroot, existing_jobs):
-        '''Return the list of running plot jobs, returning previous jobs if they still exist, and
-           new jobs intialized.  Does not update info on existing jobs, which means information in
-           the list of jobs will be from different times.  Use this when calling frequent updates,
-           and use get_running_jobs() to force a full refresh.'''
-        jobs = []
-        existing_jobs_by_pid = { j.proc.pid: j for j in existing_jobs }
+        if cached_jobs:
+            cached_jobs_by_pid = { j.proc.pid: j for j in cached_jobs }
 
         for proc in psutil.process_iter(['pid', 'cmdline']):
-            if is_plotting_cmdline(proc.cmdline()):
-                if proc.pid in existing_jobs_by_pid.keys():
-                    jobs.append(existing_jobs_by_pid[proc.pid])  # Copy from cache
-                else:
-                    jobs.append(Job(proc, logroot))
+            try:
+                if is_plotting_cmdline(proc.cmdline()):
+                    if cached_jobs and proc.pid in cached_jobs_by_pid.keys():
+                        jobs.append(cached_jobs_by_pid[proc.pid])  # Copy from cache
+                    else:
+                        jobs.append(Job(proc, logroot))
+            except psutil.NoSuchProcess:
+                # Generally this is a no-op; the process in question terminated
+                # after it was returned from the iter but before we got a
+                # chance to inspect its info.
+                pass
```

Comment:
I don't know if it actually reads better here or not, mostly due to the comment, but fyi...

https://docs.python.org/3.9/library/contextlib.html#contextlib.suppress

```suggestion
            # Ignore processes which most likely have terminated between the time of
            # iteration and data access.
            with contextlib.suppress(psutil.NoSuchProcess):
                if is_plotting_cmdline(proc.cmdline()):
                    if cached_jobs and proc.pid in cached_jobs_by_pid.keys():
                        jobs.append(cached_jobs_by_pid[proc.pid])  # Copy from cache
                    else:
                        jobs.append(Job(proc, logroot))
```